### ISSUE TITLE
How To: Dynamically set opacity of line series

### ISSUE THREAD

**Issue by KevinFernandes** [highcharts/highcharts#11750]
I am attempting to set the opacity of a line series dynamically - basically I want to "ghost" all of the series on a particular axis when the user clicks a button or ghost an individual series when the user clicks a button.

I have tried the following options
series.update({ opacity: 0.4 })
series.update({ opacity: 0.4 }, true)
series.options.opacity = 0.4 with chart.redraw();

Though I see the opacity value changing in the chart object, the chart does not actually change, as if something else is overriding the opacity when it renders.

What am i doing wrong?  Is it possible to do this dynamically?

#### Product version
Highcharts 7.1.3 with Stock module installed

**Comment by pawelfus** [comment]
Hi @KevinFernandes 

Thank you for reporting the issue!

**Workaround:**
Set `series.opacity` and `series.states.inactive.opacity` to the same value, then after an update, set `inactive` state to the series.
Demo: https://jsfiddle.net/BlackLabel/rs13qfkp/

Snippet: 
```js
  chart.series[0].update({ opacity: 0.1 });
  chart.series[0].setState('inactive');
```

*Internal note:*
 `opacity` option is applied only when series' state has changed and initial `opacity` setter is not implemented.

**Comment by KevinFernandes** [comment]
Thank you this worked! :)

In my case, the opacity needs to be toggled between "ghosted" and "normal".
The documentation for the series.setState() API would seem to imply that there are only two valid states "hover" and "undefined" (the normal state).  It took a bit of trial and error but I found that setting the state to "normal" instead of "undefined" was what I needed to restore the line properly.

I am going to enter another issue for the documentation.  

Please let me know if setting the state to "normal" is the correct way to do this (for the time being at least) or if there is a different way to accomplish this.

**Comment by pawelfus** [comment]
Yes, please create another issue for the docs, it should be there. Available states are `normal`, `hover`, `select` and `inactive`.

> Please let me know if setting the state to "normal" is the correct way to do this (for the time being at least) or if there is a different way to accomplish this.

Yes this is the correct way.

**Comment by PAPIONbit** [comment]
Have to know the way of fixing the issue #12092 can be right or any better way?

**Comment by pawelfus** [comment]
Hi @PAPIONbit 

In general, this is bug in documentation. `opacity` suppose to work only with states (was implemented with `inactive` state in v7.1) - as I said above, initial setter for `opacity` was not implemented, neither designed. Your solution looks fine.

**Comment by swiss-chris** [comment]
> Yes, please create another issue for the docs, it should be there. Available states are `normal`, `hover`, `select` and `inactive`.
> 
> > Please let me know if setting the state to "normal" is the correct way to do this (for the time being at least) or if there is a different way to accomplish this.
> 
> Yes this is the correct way.



> Yes, please create another issue for the docs, it should be there. Available states are `normal`, `hover`, `select` and `inactive`.
> 
> > Please let me know if setting the state to "normal" is the correct way to do this (for the time being at least) or if there is a different way to accomplish this.
> 
> Yes this is the correct way.

Could someone please update this in the official documentation? It took me at least an hour to find the solution to my problem here.
